src/System.Net.Http/tests/UnitTests/HttpSystemProxyTest.cs

```diff
@@ -40,5 +40,44 @@ public void HttpProxy_SystemProxy_Loaded()
             Assert.Equal(fakeProxyUri, p.GetProxy(fooHttp));
             Assert.Equal(fakeProxyUri, p.GetProxy(fooHttps));
         }
+
+        [Theory]
+        [InlineData("http://localhost/", true)]
+        [InlineData("http://127.0.0.1/", true)]
+        [InlineData("http://128.0.0.1/", false)]
+        [InlineData("http://[::1]/", true)]
+        [InlineData("http://foo/", true)]
+        [InlineData("http://www.foo.com/", true)]
+        [InlineData("http://WWW.FOO.COM/", true)]
+        [InlineData("http://foo.com/", false)]
+        [InlineData("http://bar.com/", true)]
+        [InlineData("http://BAR.COM/", true)]
+        [InlineData("http://162.1.1.1/", true)]
+        [InlineData("http://[2a01:5b40:0:248::52]/", false)]
+        [InlineData("http://[2002::11]/", true)]
+        [InlineData("http://[2607:f8b0:4005:80a::200e]/", true)]
+        [InlineData("http://[2607:f8B0:4005:80A::200E]/", true)]
+        public void HttpProxy_Local_Bypassed(string name, bool shouldBypass)
+        {
+
+            RemoteInvoke((url, expected) =>
```

Comment:
>RemoteInvoke [](start = 12, length = 12)

Q: why RemoteInvoke is needed?

Reply:
as far as I can tell the fake registry is global - similar way how the environment work for Linux. 
I did not want tests to impact each other and I did not find better way how to do that.